salt/modules/file.py

```diff
@@ -920,13 +920,12 @@ def uncomment(path,
 
         salt '*' file.uncomment /etc/hosts.deny 'ALL: PARANOID'
     '''
-    # Largely inspired by Fabric's contrib.files.uncomment()
-
-    return sed(path,
-               before=r'^([[:space:]]*){0}'.format(char),
-               after=r'\1',
-               limit=regex.lstrip('^'),
-               backup=backup)
+    pattern = '^{0}{1}'.format(char, regex.lstrip('^').rstrip('$'))
+    repl = "{0}".format(regex.lstrip('^').rstrip('$'))
```

Comment:
This is completely wrong! You are replacing the matched pattern with the regex itself.
This can only work if you don't actually use a regular expression.

For example if you try to uncomment `level = (warning|error)` and the file contains `# level = warning` it will be replaced with `level = (warning|error)`.